test/e2e/upgradetest/resize_test.go

```diff
@@ -29,6 +29,9 @@ func TestResize(t *testing.T) {
 		if err != nil {
 			t.Fatal(err)
 		}
+		if err = e2eutil.WaitUntilDeploymentDeleted(testF.KubeCli, 3*time.Second, 30*time.Second, "etcd-operator", testF.KubeNS); err != nil {
```

Comment:
Okay changing the grace period to zero resolved the conflict.
It seems the deployment goes away immediately(as far as I could see) but the pod takes some more time to delete.
For the purposes of this test since the pod names are generated they don't conflict. Only the deployment needs to be gone before the next test.